gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -378,41 +392,79 @@ def run(self) -> GtsfmData:
+        # Group results by subgraph
+        subgraph_two_view_results = group_results_by_subgraph(two_view_results_dict, subgraphs)
 
-        assert isinstance(sfm_result, GtsfmData)
-        all_metrics_groups.extend(mvo_metrics_groups)
+        # Process each subgraph
+        all_delayed_sfm_results = []
+        all_delayed_io = []
+        all_delayed_mvo_metrics_groups = []
+
+        for idx, subgraph_result_dict in enumerate(subgraph_two_view_results):
+            logger.info(f"Creating computation graph for subgraph {idx+1}/{len(subgraph_two_view_results)} with {len(subgraph_result_dict)} image pairs")
+            
+            # Unzip the two-view results for this subgraph
+            subgraph_i2Ri1_dict, subgraph_i2Ui1_dict, subgraph_v_corr_idxs_dict, _, subgraph_post_isp_reports = unzip_two_view_results(
+                subgraph_result_dict
+            )
+            
+            # Create computation graph for this subgraph
+            if len(subgraph_i2Ri1_dict) > 0:  # Only process non-empty subgraphs
+                delayed_sfm_result, delayed_io, delayed_mvo_metrics_groups = self.scene_optimizer.create_computation_graph(
+                    keypoints_list=keypoints_list,
+                    i2Ri1_dict=subgraph_i2Ri1_dict,
+                    i2Ui1_dict=subgraph_i2Ui1_dict,
+                    v_corr_idxs_dict=subgraph_v_corr_idxs_dict,
+                    two_view_reports=subgraph_post_isp_reports,
+                    num_images=len(self.loader),
+                    images=self.loader.create_computation_graph_for_images(),
+                    camera_intrinsics=intrinsics,
+                    relative_pose_priors=self.loader.get_relative_pose_priors(list(subgraph_i2Ri1_dict.keys())),
+                    absolute_pose_priors=self.loader.get_absolute_pose_priors(),
+                    cameras_gt=self.loader.get_gt_cameras(),
+                    gt_wTi_list=self.loader.get_gt_poses(),
+                    gt_scene_mesh=self.loader.get_gt_scene_trimesh(),
+                )
+                all_delayed_sfm_results.append(delayed_sfm_result)
+                all_delayed_io.extend(delayed_io)
+                all_delayed_mvo_metrics_groups.extend(delayed_mvo_metrics_groups)
+
+        # Compute all the delayed objects
+        with performance_report(filename="scene-optimizer-dask-report.html"):
+            if all_delayed_sfm_results:
+                results = dask.compute(*all_delayed_sfm_results, *all_delayed_io, *all_delayed_mvo_metrics_groups)
+                sfm_results = results[:len(all_delayed_sfm_results)]
+                other_results = results[len(all_delayed_sfm_results):]
+                
+                # Extract metrics from results
+                mvo_metrics_groups = [x for x in other_results if isinstance(x, GtsfmMetricsGroup)]
+                all_metrics_groups.extend(mvo_metrics_groups)
+                
+                # For now, return the first non-empty result
```

Comment:
you can also assert that there is only one result since we only support one right now. but this works.